pypeit/core/flat.py

```diff
@@ -1,488 +1,491 @@
+        new_left += left_shift * slitwidth
+
+    # ------------------------------------------------------------------
+    # Adjust the right edge
+
+    # Get the maximum to the right of the center
+    # TODO: Set a parameter for this
+    iright = (spat_coo > 0.6) & (spat_coo < 0.9)
+    if not np.any(iright):
+        msgs.error('No coordinates toward the right of the slit center.  Slit boundaries are '
+                   'likely in error, and you probably have a bad (very short) slit.  Slit center '
+                   'at center row is {0:.1f}.'.format((left[nspec//2] + right[nspec//2])/2))
+    right_thresh = thresh * np.amax(norm_flat[iright])
+
+    # Find the data that are less than the provided threshold and
+    # within the limits set by the offset
+    indx = (spat_coo > (1-maxfrac)) & (norm_flat < right_thresh)
+
+    # To tweak, there must be at least one pixel that meets the above
+    # criteria
+    right_shift = 0.
+    new_right = np.copy(right)
+    if np.any(indx):
+        # Find the first index
+        i = np.where(indx)[0][0]
+        if i < norm_flat.size-1 and norm_flat[i+1] > norm_flat[i]:
+            # TODO: Not sure what to do here.  Check if this ever happens...
+            msgs.error('Flat is noisy!  Faulting...')
+        if norm_flat[i-1] < right_thresh:
+            msgs.warn('Right slit boundary tweak limited by maximum allowed shift: {:.1f}%'.format(
+                        100*maxfrac))
+            right_shift = maxfrac
+        else:
+            right_shift = 1-linear_interpolate(norm_flat[i-1], spat_coo[i-1], norm_flat[i],
+                                               spat_coo[i], right_thresh)
+        msgs.info('Tweaking right slit boundary by {0:.1f}%'.format(100*right_shift) +
+                  ' % ({0:.2f} pixels)'.format(right_shift*slitwidth))        
+        new_right -= right_shift * slitwidth
 
-    # Compute all the final output images output
-    pixelflat = np.ones_like(flat)
-    flat_model = np.ones_like(flat)
-    flat_model[thismask_out] = twod_model[thismask_out]*np.fmax(illumflat[thismask_out],0.05)*np.fmax(spec_model[thismask_out],1.0)
-    pixelflat[thismask_out] = flat[thismask_out]/flat_model[thismask_out]
+    return left_thresh, left_shift, new_left, right_thresh, right_shift, new_right
 
-    # ToDo Add some code here to treat the edges and places where fits go bad?
-    # Set the pixelflat to 1.0 wherever the flat was nonlinear
-    pixelflat[flat >= nonlinear_counts] = 1.0
-    # Do not apply pixelflat field corrections that are greater than 100% to avoid creating edge effects, etc.
-    # TODO Should we do the same for the illumflat??
-    #pixelflat = np.fmax(np.fmin(pixelflat, 2.0), 0.5)
-    pixelflat = np.clip(pixelflat, 0.5, 2.0)
 
-    return pixelflat, illumflat, flat_model, tilts, thismask_out, slit_left_out, slit_righ_out
+# TODO: This is deprecated, and replaced by tweak_slit_edges above.
```

Comment:
Please make a module in deprecated for these old codes, including the old version of the flat fielding code.

Reply:
Will do.